src/main.cpp

```diff
@@ -26,14 +26,37 @@ bool isInRecoveryMode() {
   }
   saveSettings();
   DPRINTF("Wrote QPC = %d\n", Settings->quick_power_cycle_count);
+  // making sure the board can't go into a locked state
+  Settings->board_locked = false;
   return recovery;
+  return false;
+}
+
+bool isLocked() {
```

Comment:
isLocked should have nothing to to with recovery mode. Make this function void and rename to something like "manageLockedStateTransitionOnStartup()" the more verbose, the better ;)

Add a function-level comment outlining that after running this function, Settings->board_locked is an authoritative source of whether or not the board is locked.